### PR TITLE
LPD-19824 added new API endpoints for delivery cart

### PR DESCRIPTION
**Pull request by brbalazs**
https://liferay.atlassian.net/browse/LPD-19824

### CODE REVIEW

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-impl/rest-openapi.yaml [hunk 1]
```diff
@@ -573,25 +573,487 @@ info:
...
+                      minLength: 1
+                      type: string
                   style: simple
             responses:
                 204:
-                    content:
-                        application/json: {}
-                        application/xml: {}
+                    description:
+                        No content
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-impl/rest-openapi.yaml:1056]: why the response of this delete is No content compared with the byId where it's application/json: {}

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-test/src/testIntegration/java/com/liferay/headless/commerce/delivery/cart/resource/v1_0/test/CartItemResourceTest.java [hunk 2]
```diff
@@ -202,6 +266,21 @@ protected CartItem testPutCartItem_addCartItem() throws Exception {
...
+			_commerceOrder.getExternalReferenceCode(), randomCartItem());
+	}
+
+	@Override
+	protected CartItem testPutCartItemByExternalReferenceCode_createCartItem()
+		throws Exception {
+
+		return randomCartItem();
+	}
+
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-test/src/testIntegration/java/com/liferay/headless/commerce/delivery/cart/resource/v1_0/test/CartItemResourceTest.java:283]: Seems to be the same as in the base file. I think not needed

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-test/src/testIntegration/java/com/liferay/headless/commerce/delivery/cart/resource/v1_0/test/CartResourceTest.java [hunk 3]
```diff
@@ -228,6 +228,31 @@ protected Long testGetChannelCartsPage_getChannelId() throws Exception {
 		return _commerceChannel.getCommerceChannelId();
 	}
 
+	@Override
+	protected Cart testGetChannelCartsPageByExternalReferenceCode_addCart(
+			String accountExternalReferenceCode,
+			String channelExternalReferenceCode, Cart cart)
+		throws Exception {
+
+		return cartResource.postCartCheckout(cart.getId());
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-test/src/testIntegration/java/com/liferay/headless/commerce/delivery/cart/resource/v1_0/test/CartResourceTest.java:237]: doesn't exist the cartResource.postCartByExternalReferenceCodeCheckout?

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-test/src/testIntegration/java/com/liferay/headless/commerce/delivery/cart/resource/v1_0/test/CartResourceTest.java [hunk 4]
```diff
@@ -287,6 +312,13 @@ protected Cart testPostCartCouponCode_addCart(Cart cart) throws Exception {
 		return cartResource.postCartCouponCode(cart.getId(), couponCode);
 	}
 
+	@Override
+	protected Cart testPostChannelCartByExternalReferenceCode_addCart(Cart cart)
+		throws Exception {
+
+		return cartResource.postCartCheckout(cart.getId());
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-test/src/testIntegration/java/com/liferay/headless/commerce/delivery/cart/resource/v1_0/test/CartResourceTest.java:319]: doesn't exist the cartResource.postCartByExternalReferenceCodeCheckout?

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by brbalazs** [comment]
ci:test:sf

**Comment by brbalazs** [comment]
ci:test:sf

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-impl/src/main/java/com/liferay/headless/commerce/delivery/cart/internal/resource/v1_0/CartCommentResourceImpl.java [hunk 5]
```diff
@@ -44,6 +45,24 @@ public void deleteCartComment(Long commentId) throws Exception {
...
+			String externalReferenceCode)
+		throws Exception {
+
+		CommerceOrderNote commerceOrderNote =
+			_commerceOrderNoteService.fetchByExternalReferenceCode(
+				externalReferenceCode, contextCompany.getCompanyId());
+
+		if (commerceOrderNote == null) {
+			throw new NoSuchOrderNoteException(
+				"Unable to find order comment with external reference code " +
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-impl/src/main/java/com/liferay/headless/commerce/delivery/cart/internal/resource/v1_0/CartCommentResourceImpl.java:59]: Unable to find order note

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-impl/src/main/java/com/liferay/headless/commerce/delivery/cart/internal/resource/v1_0/CartCommentResourceImpl.java [hunk 6]
```diff
@@ -75,6 +94,24 @@ public CartComment getCartComment(Long commentId) throws Exception {
...
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-impl/src/main/java/com/liferay/headless/commerce/delivery/cart/internal/resource/v1_0/CartCommentResourceImpl.java:108]: Unable to find order note

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-impl/src/main/java/com/liferay/headless/commerce/delivery/cart/internal/resource/v1_0/CartCommentResourceImpl.java [hunk 7]
```diff
@@ -132,6 +169,25 @@ public CartComment putCartComment(Long commentId, CartComment cartComment)
...
+			String externalReferenceCode, CartComment cartComment)
+		throws Exception {
+
+		CommerceOrderNote commerceOrderNote =
+			_commerceOrderNoteService.fetchByExternalReferenceCode(
+				externalReferenceCode, contextCompany.getCompanyId());
+
+		if (commerceOrderNote == null) {
+			throw new NoSuchOrderNoteException(
+				"Unable to find order comment with external reference code " +
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-impl/src/main/java/com/liferay/headless/commerce/delivery/cart/internal/resource/v1_0/CartCommentResourceImpl.java:183]: Unable to find order note

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by brbalazs** [comment]
ci:test:sf

**Review by smottal — APPROVED** [review]

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:commerce-functional

**Comment by smottal** [comment]
ci:test:sf

**Comment by brbalazs** [comment]
ci:test:sf

**Comment by brbalazs** [comment]
ci:test:relevant

**Comment by brbalazs** [comment]
ci:test:commerce-functional

**Comment by brbalazs** [comment]
ci:test:sf

**Comment by brbalazs** [comment]
ci:test:sf

**Comment by brbalazs** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:forward

**Pull request closed without merging**